src/random_test.cpp

```diff
@@ -1098,9 +1098,15 @@ void Table::Check(Thd1 *thd) {
     int partition =
         rand_int(static_cast<Partition *>(this)->number_of_part - 1);
     table_mutex.unlock();
-    execute_sql("ALTER TABLE " + name_ + " CHECK PARTITION p" +
-                    std::to_string(partition),
-                thd);
+    auto row = mysql_fetch_row(get_result("ALTER TABLE " + name_ +
```

Comment:
Here, we're just picking a random number and using that to pick a random partition name and executing our query which remains the same for all kinds of partitions. So, I don't think checking the type of partition here would add any value. 

We could use the contents of the step_X.ddl to read a random partition name instead of generating using random_int() which would reduce the number of errors in my opinion. What do you think?